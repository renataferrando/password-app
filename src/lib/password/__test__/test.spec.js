import React from "react";

import { render, screen, fireEvent } from "@testing-library/react";
import Component from "../index";

const defaultProps = {
  number: false,
  character: false,
  uppercase: false,
  noConsecutive: false,
};

describe("Given Password component", () => {
  test("Should render as expected", () => {
    const { container } = render(<Component {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  test("When numberProp is set to true should show requirement", () => {
    const testProps = {
      ...defaultProps,
      number: true
    }
    render(
      <Component {...testProps}></Component>);
    const numberRequirement = screen.queryByText(/Has a number 0-9/i);
    expect(numberRequirement).toBeInTheDocument();
  });

  test('When type on input it should ....', () => {
    render(
      <Component {...defaultProps}></Component>);
      const input = screen.getByRole('textbox');
      fireEvent.change(input, { target: { value: 'Za' } });
      expect(input.value).toBe('Za')    
  })
});
