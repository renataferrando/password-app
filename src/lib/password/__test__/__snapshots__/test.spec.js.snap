// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Password component Should render as expected 1`] = `
<div>
  <div
    class="passwordWrapper"
  >
    <h2>
      Password component
    </h2>
    <div
      class="password"
    >
      <input
        id="name"
        name="name"
        type="text"
        value=""
      />
      <div
        class="validation"
      >
        <div
          class="checkmarkWrapper"
          role="helper"
        >
          <span
            class="checkmarkFalse"
          >
            x
          </span>
          <span>
            Has a number 0-9
          </span>
        </div>
        <div
          class="checkmarkWrapper"
          role="helper"
        >
          <span
            class="checkmarkFalse"
          >
            x
          </span>
          <span>
            Has a special char !@#$%^&*
          </span>
        </div>
        <div
          class="checkmarkWrapper"
          role="helper"
        >
          <span
            class="checkmarkFalse"
          >
            x
          </span>
          <span>
            Has uppercase Letter
          </span>
        </div>
        <div
          class="checkmarkWrapper"
          role="helper"
        >
          <span
            class="checkmarkFalse"
          >
            x
          </span>
          <span />
        </div>
      </div>
    </div>
  </div>
</div>
`;
